autograd/core.py

```diff
@@ -216,7 +216,7 @@ def _inner_prod(self, x, y): assert False
     def _covector(self, x):      return x
 
     def __eq__(self, other):
-        return type(self) == type(other) and self.__dict__ == other.__dict__
+        return isinstance(self, type(other)) and self.__dict__ == other.__dict__
```

Comment:
These changes not only affect the meaning of the code (`isinstance` will return true for subclasses), they also affect its performance, since `isinstance` is often significantly slower than `type(...) is ...`, so I can't accept these.